evaluators/text/BleuEvaluator/__init__.py

```diff
@@ -1,38 +1,38 @@
-from jina.executors.evaluators import BaseEvaluator
-from typing import Sequence, Any
 import nltk.translate.bleu_score as bleu
-import numpy as np
-from nltk import ngrams
-from collections import Counter
-
 
+from jina.executors.evaluators import BaseEvaluator
 
 class BleuEvaluator(BaseEvaluator):
     """
     :class:`BleuEvaluator`Bilingual Evaluation Understudy Score. 
-    Evaluates the generated sentence (hypothesis) against a reference sentence. 
+    Evaluates the generated sentence (actual) against a desired sentence. 
     It will use the Bleu on NLTK package.
     A perfect match will score 1.0 and a perfect unmatch will score 0.0
     """
     
     
-    def get_nltk_bleu_score(self, hypothesis, reference):
-        return bleu.sentence_bleu(reference, hypothesis)
+    def get_nltk_bleu_score(self, desired, actual):
+        return bleu.sentence_bleu(desired, actual)
 
 
     def evaluate(self,
-            hypothesis,
-            reference,
-            nltk_default = True,
+            desired,
+            actual, 
             *args,
             **kwargs) -> float:
         """"
-        :param actual: the scores predicted by the search system.
         :param desired: the expected score given by user as groundtruth.
+        :param actual: the scores predicted by the search system.
         :return the evaluation metric value for the request document.
-        """
 
-        BLEUScore = self.get_nltk_bleu_score(hypothesis, reference)
-
-        return BLEUScore
+        NLTK expectes an array of strings, 
+        so the incoming string needs to be tokenized first.
+        They will be stored in a desired_list
+        """
+        
+        # tokenize
+        desired_list = desired.split()
+        actual_list = actual.split()
+        
+        return self.get_nltk_bleu_score([desired_list], actual_list)
```

Comment:
do you need to make a `list` of a `list`?

Reply:
Yep, cause it can be a list of many references, and each of them is a list of strings (one str per word)

Reply:
Ok!